src/browzine-summon-adapter.js

```diff
@@ -1,30 +1,52 @@
-// Begin BrowZine-Summon Integration Code
-angular.module("summonApp.directives")
-.constant("api", "https://api.thirdiron.com/public/v1/libraries/XXX")
-.constant("apiKey", "ENTER API KEY")
-.constant("bookIcon", "https://s3.amazonaws.com/thirdiron-assets/images/integrations/browzine_open_book_icon.png")
-.directive("documentSummary", ["$http", "$sce", "api", "apiKey", "bookIcon", function(http, sce, api, apiKey, bookIcon) {
+browzine.search = (function() {
+  var api = browzine.api;
+  var apiKey = browzine.apiKey;
+
   function isArticle(data) {
-    if(typeof data.document !== "undefined" && data.document !== null) {
-      return data.document.content_type.trim() === "Journal Article";
-    } else {
-      return data.type === "articles";
+    var result = false;
+
+    if(data.document) {
+      if(data.document.content_type) {
+        var contentType = data.document.content_type.trim();
+        result = (contentType === "Journal Article");
+      }
+    }
+
+    if(data.type) {
+      result = (data.type === "articles");
     }
+
+    return result;
   };
 
   function isJournal(data) {
-    if(typeof data.document !== "undefined" && data.document !== null) {
-      return data.document.content_type.trim() === "Journal";
-    } else {
-      return data.type === "journals";
+    var result = false;
+
+    if(data.document) {
+      if(data.document.content_type) {
+        var contentType = data.document.content_type.trim();
+        result = (contentType === "Journal" || contentType === "eJournal");
+      }
+    }
+
+    if(data.type) {
+      result = (data.type === "journals");
```

Comment:
Same thing here - what if we just looked if `journal` appeared in `data.type` ?

Of course, maybe that's just getting things to be too loose... not sure, just trying to reduce the amount of future changes we may have to do. But if some other library has a type of, say, "ejournals", eh, it's pretty easy to change our logic, so maybe no change should be done here. Just thinking out loud 😊

Reply:
Yeah, the key is getting as much customers as possible using the dynamic script, then enhancements and bug fixes are as simple as a pull request and boom they're productive again.